src/Ssh/Ssh/Common/BastionUtils.cs

```diff
@@ -0,0 +1,318 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+
+
+    using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+    using Microsoft.Azure.PowerShell.Ssh.Helpers.Network;
+
+
+    using Microsoft.Azure.PowerShell.Ssh.Helpers.Network.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Ssh.AzureClients;
+    using Microsoft.Azure.PowerShell.Ssh.Helpers.Compute;
+    using Newtonsoft.Json;
+
+    
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using System.Linq;
+    using Microsoft.Rest.Azure;
+    using System;
+    using System.Management.Automation;
+    using System.Net;
+    using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Commands.Common.Exceptions;
+    using System.Reflection;
+    using Microsoft.Azure.PowerShell.Cmdlets.Ssh.Common;
+    using Microsoft.Azure.Management.WebSites.Version2016_09_01.Models;
+    using System.Net.Http;
+    using System.Text;
+    using System.Threading;
+    using Newtonsoft.Json.Linq;
+    using System.Management.Automation.Language;
+    using System.Diagnostics;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.Ssh.Common
+
+    {
+    internal class BastionUtils
+    {
+        #region Fields
+        private NetworkClient _networkClient;
+        private IAzureContext _context;
+        #endregion
+
+        public BastionUtils(IAzureContext context)
+        {
+            _context = context;
+        }
+
+        #region Properties
+        private NetworkClient NetworkClient
+        {
+            get
+            {
+                if (_networkClient == null)
+                {
+                    _networkClient = new NetworkClient(_context);
+                }
```

Comment:
You can simplify this to:

```suggestion
                _networkClient ??= new NetworkClient(_context);
```